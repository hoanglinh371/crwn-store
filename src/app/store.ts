import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';

import cartReducer from '../features/cart/redux/cart.slice';
import categoryReducer from '../features/category/redux/category.slice';
import productReducer from '../features/product/redux/product.slice';

export const store = configureStore({
  reducer: {
    productReducer,
    cartReducer,
    categoryReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
