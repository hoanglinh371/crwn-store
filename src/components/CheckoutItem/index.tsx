import React from 'react';
import classNames from 'classnames/bind';

import styles from './CheckoutItem.module.scss';

interface Props {
    cartItem: any;
}

const cx = classNames.bind(styles);

const CheckoutItem: React.FC<Props> = ({ cartItem }) => {
    const { name, imageUrl, price, quantity } = cartItem;

    return (
        <div className={cx('container')}>
            <div className={cx('image-container')}>
                <img src={imageUrl} alt={name} />
            </div>
            <span className={cx('name')}>{name}</span>
            <span className={cx('quantity')}>
                <div className={cx('arrow')}>&#10094;</div>
                <span className={cx('value')}>{quantity}</span>
                <div className={cx('arrow')}>&#10095;</div>
            </span>
            <span className={cx('price')}>{price}</span>
            <div className={cx('remove-button')}>&#10005;</div>
        </div>
    );
};

export default CheckoutItem;
