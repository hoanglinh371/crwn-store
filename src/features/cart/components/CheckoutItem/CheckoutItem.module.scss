.container {
  width: 100%;
  display: flex;
  min-height: 100px;
  border-bottom: 1px solid darkgrey;
  padding: 16px 0;
  font-size: 1rem;
  align-items: center;

  .image-container {
    width: 23%;
    padding-right: 16px;

    img {
      width: 100%;
      height: 100%;
    }
  }
  .name,
  .quantity,
  .price {
    width: 23%;
  }

  .quantity {
    display: flex;
    align-items: center;

    .arrow {
      cursor: pointer;
    }

    .value {
      margin: 0 10px;

      @media screen and (max-width: 600px) {
        margin: 0;
      }
    }
  }

  .remove-button {
    cursor: pointer;
  }
}
