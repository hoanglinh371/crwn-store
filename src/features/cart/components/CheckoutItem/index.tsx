import React from 'react';
import classNames from 'classnames/bind';
import { FaChevronLeft, FaChevronRight } from 'react-icons/fa';
import { IoClose } from 'react-icons/io5';
import { useAppDispatch } from 'src/app/hooks';

import {
  addProduct,
  removeProduct,
  clearProduct,
} from 'src/features/cart/redux/cart.slice';
import { CartItem as CartItemProps } from 'src/features/cart/redux/cart.types';
import styles from './CheckoutItem.module.scss';

interface Props {
  cartItem: CartItemProps;
}

const cx = classNames.bind(styles);

const CheckoutItem: React.FC<Props> = ({ cartItem }) => {
  const { name, imageUrl, price, quantity } = cartItem;
  const dispatch = useAppDispatch();

  const handleAddProduct = () => dispatch(addProduct(cartItem));

  const handleRemoveProduct = () => dispatch(removeProduct(cartItem));

  const handleClearProduct = () => dispatch(clearProduct(cartItem));

  return (
    <div className={cx('container')}>
      <div className={cx('image-container')}>
        <img src={imageUrl} alt={name} />
      </div>
      <span aria-label='name' className={cx('name')}>
        {name}
      </span>
      <span className={cx('quantity')}>
        <div
          aria-label='dec-btn'
          className={cx('arrow')}
          onClick={handleRemoveProduct}
        >
          <FaChevronLeft />
        </div>
        <span className={cx('value')}>{quantity}</span>
        <div
          aria-label='inc-btn'
          className={cx('arrow')}
          onClick={handleAddProduct}
        >
          <FaChevronRight />
        </div>
      </span>
      <span className={cx('price')}>${price}</span>
      <div
        aria-label='del-btn'
        className={cx('remove-button')}
        onClick={handleClearProduct}
      >
        <IoClose />
      </div>
    </div>
  );
};

export default CheckoutItem;
