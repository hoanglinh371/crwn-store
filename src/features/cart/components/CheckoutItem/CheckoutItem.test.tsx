import { cleanup, render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import { store } from 'src/app/store';
import { CartItem as CartItemProps } from 'src/interfaces/product.interface';
import CheckoutItem from '.';

const cartItem: CartItemProps = {
  id: 1,
  name: 'Hoodie',
  imageUrl: 'hoodie.png',
  price: 250,
  quantity: 1,
};

describe('CheckoutItem component', () => {
  afterEach(cleanup);

  it('Should_MatchSnapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <CheckoutItem cartItem={cartItem} />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should_HaveName', () => {
    render(
      <Provider store={store}>
        <CheckoutItem cartItem={cartItem} />
      </Provider>,
    );
    expect(screen.getByLabelText('name')).toHaveTextContent('Hoodie');
  });
});
