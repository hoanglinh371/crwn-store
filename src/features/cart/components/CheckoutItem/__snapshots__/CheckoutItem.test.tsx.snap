// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem component Should_MatchSnapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="image-container"
    >
      <img
        alt="Hoodie"
        src="hoodie.png"
      />
    </div>
    <span
      aria-label="name"
      class="name"
    >
      Hoodie
    </span>
    <span
      class="quantity"
    >
      <div
        aria-label="dec-btn"
        class="arrow"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
          />
        </svg>
      </div>
      <span
        class="value"
      >
        1
      </span>
      <div
        aria-label="inc-btn"
        class="arrow"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 320 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          />
        </svg>
      </div>
    </span>
    <span
      class="price"
    >
      $
      250
    </span>
    <div
      aria-label="del-btn"
      class="remove-button"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
        />
      </svg>
    </div>
  </div>
</div>
`;
